subseas_flex_fcst/cpt.py

```diff
@@ -0,0 +1,95 @@
+import glob
+import re
+from datetime import datetime
+import numpy as np
+import cptio
+import xarray as xr
+
+#select file for specific lead time and start date
+def sel_file(data_path, filename_pattern, lead_time, start_date, format_in):
+    """ Select a single cpt file for a given start and lead.
+
+    Parameters
+    ----------
+    data_path : str
+        String of the path pointing to cpt datasets.
+    filename_pattern : str
+        String of the filename pattern name for a given variable's data file.
+    lead_time : str
+         String of the lead time value to be selected for as is represented in the file name.
+    start_date : str
+        String of the start date to be selected for as is represented in the file name.
+    format_in: str
+        String of the strftime format of `start_date` in the `filename_pattern` .
+    Returns
+    -------
+    file_selected : xarray Dataset
+        Single CPT data file as multidimensional xarray dataset.
+    Notes
+    -----
+    `filename_pattern` should be most common denominator for any group of datasets,
+    so that a single file can be selected using only `lead_time` and `start_date`.
+    Examples
+    --------
+    For files which have naming structure such as the example file:
+        CFSv2_SubXPRCP_CCAFCST_mu_Apr_Apr-1-2022_wk1.txt
+    And where this file's `lead_time` and `start_date`:
+        `lead_time` == 'wk1' and `start_date` == 'Apr-1-2022'
+    `filename_pattern` == 'CFSv2_SubXPRCP_CCAFCST_mu_Apr_*.txt'
+    """
+    pattern = f"{start_date}_{lead_time}"
+    full_path = f"{data_path}/{filename_pattern}"
+    expanded_name = glob.glob(full_path.replace("mystartandlead",pattern))
+    if len(expanded_name) == 0:
+        file_selected = xr.Dataset().merge(xr.DataArray(np.nan).rename("missing_file"))
+    else:
+        file_name = expanded_name[0]
+        file_selected = cptio.open_cptdataset(file_name)
```

Comment:
Misleading variable name--I think it's a DataArray or Dataset, not a file?

Reply:
renamed along with function in last commit